rotkehlchen/chain/ethereum/modules/pickle/pickle.py

```diff
@@ -0,0 +1,147 @@
+from typing import Any, Dict, List, NamedTuple, Optional, TYPE_CHECKING
+
+from rotkehlchen.accounting.structures import AssetBalance, Balance
+from rotkehlchen.chain.ethereum.contracts import EthereumContract
+from rotkehlchen.chain.ethereum.utils import multicall_2, token_normalized_value_decimals
+from rotkehlchen.constants.assets import A_PICKLE
+from rotkehlchen.constants.ethereum import PICKLE_DILL, PICKLE_DILL_REWARDS
+from rotkehlchen.errors import DeserializationError
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.utils.interfaces import EthereumModule
+from rotkehlchen.premium.premium import Premium
+from rotkehlchen.serialization.deserialize import deserialize_timestamp
+from rotkehlchen.typing import ChecksumEthAddress, Timestamp
+from rotkehlchen.user_messages import MessagesAggregator
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.manager import EthereumManager
+    from rotkehlchen.db.dbhandler import DBHandler
+
+
+class DillBalance(NamedTuple):
+    dill_amount: AssetBalance
+    pending_rewards: AssetBalance
+    lock_time: Timestamp
+
+    def serialize(self) -> Dict[str, Any]:
+        return {
+            "locked_amount": self.dill_amount.serialize(),
+            "pending_rewards": self.pending_rewards.serialize(),
+            "locked_until": self.lock_time,
+        }
+
+
+class PickleFinance(EthereumModule):
+
+    def __init__(
+            self,
+            ethereum_manager: 'EthereumManager',
+            database: 'DBHandler',
+            premium: Optional[Premium],
+            msg_aggregator: MessagesAggregator,
+    ) -> None:
+        self.ethereum = ethereum_manager
+        self.database = database
+        self.premium = premium
+        self.msg_aggregator = msg_aggregator
+
+    def get_dill(
```

Comment:
I would rename this to get dill_info or get_dill_balances. The name is not descriptive enough yet.